addons/mrp_subcontracting/models/stock_move.py

```diff
@@ -25,6 +26,17 @@ def _compute_show_subcontracting_details_visible(self):
             else:
                 move.show_subcontracting_details_visible = False
 
+    def _compute_show_serial_mass_receipt(self):
+        """ Compute if the action button for serial mass receipt is visible """
+        self.show_serial_mass_receipt = False
+        for move in self:
+            if move.state not in ('draft', 'cancel', 'done') and move.is_subcontract and move.product_id.tracking == "serial" \
+                    and float_compare(move.product_uom_qty, 1, precision_rounding=move.product_uom.rounding) > 0:
+                production = move.move_orig_ids.production_id
+                have_serial_components, have_lot_components, dummy, multiple_lot_components = production._check_serial_mass_produce_components()
+                if not have_serial_components and have_lot_components and not multiple_lot_components:
```

Comment:
Is there a reason why subcontracting requires at least one of its components to be lot tracked? I would think you could mass assign SNs even if none of its components are tracked, but maybe I am missing something?

Reply:
Mass Receipt bypasses Record Components by using standard mechanism when no serial or multiple lot component.
If no tracked component at all, another button doing the same serves nothing